components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorTestCase.java

```diff
@@ -303,12 +321,79 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
             }
         }).when(mockAuthnCtxt).setRememberMe(anyBoolean());
 
+        mockLog = mock(Log.class);
+        enableDebugLogs(mockLog, Boolean.parseBoolean(debugEnabled));
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+
+                debugMsg = (String) invocation.getArguments()[0];
+                return null;
+            }
+        }).when(mockLog).debug(anyString());
+
         basicAuthenticator.processAuthenticationResponse(mockRequest, mockResponse, mockAuthnCtxt);
 
-        assertEquals(authenticatedUser.getAuthenticatedSubjectIdentifier(), dummyUserName);
+        if (userNameValue
+                != null && userNameValue.trim().length() > 0) {
+            assertEquals(authenticatedUser.getAuthenticatedSubjectIdentifier(), userNameValue +
+                    CarbonConstants.DOMAIN_SEPARATOR + dummyDomainName + "@" + "dummyTenantDomain");
+        } else {
+            assertEquals(authenticatedUser.getAuthenticatedSubjectIdentifier(), dummyUserName);
+        }
         if (Boolean.valueOf(chkRemember)) {
             assertEquals(isrememberMe, (Boolean) true);
         }
+        if (Boolean.parseBoolean(debugEnabled) && userNameUri != null && userNameUri.trim().length() > 0) {
+            if (Boolean.valueOf(multipleAttributeEnable) && userNameValue != null &&
+                    userNameValue.trim().length() > 0) {
+                assertEquals(debugMsg, "UserNameAttribute is found for user. Value is :  " + userNameValue +
+                        CarbonConstants.DOMAIN_SEPARATOR + dummyDomainName + "@" + "dummyTenantDomain");
```

Comment:
is this correct?

dummyDomainName + CarbonConstants.DOMAIN_SEPARATOR + userNameValue + "@" + "dummyTenantDomain" ?